matchers/be_numerically_matcher.go

```diff
@@ -13,11 +13,25 @@ type BeNumericallyMatcher struct {
 }
 
 func (matcher *BeNumericallyMatcher) FailureMessage(actual interface{}) (message string) {
-	return format.Message(actual, fmt.Sprintf("to be %s", matcher.Comparator), matcher.CompareTo[0])
+	return matcher.FormatFailureMessage(actual, true)
 }
 
 func (matcher *BeNumericallyMatcher) NegatedFailureMessage(actual interface{}) (message string) {
-	return format.Message(actual, fmt.Sprintf("not to be %s", matcher.Comparator), matcher.CompareTo[0])
+	return matcher.FormatFailureMessage(actual, false)
+}
+
+func (matcher *BeNumericallyMatcher) FormatFailureMessage(actual interface{}, positive bool) (message string) {
+	var formatted string
+	if len(matcher.CompareTo) == 1 {
+		formatted = fmt.Sprintf("to be %s", matcher.Comparator)
+	} else {
+		formatted = fmt.Sprintf("to be within %v of %s", matcher.CompareTo[1], matcher.Comparator)
+	}
+	if !positive {
+		formatted = "not " + formatted
+	}
+	formatted = format.Message(actual, formatted, matcher.CompareTo[0])
```

Comment:
Can we just return here in place of setting the `formatted` variable and then returning?